backend/battle/views.py

```diff
@@ -9,12 +13,51 @@
 # from .battles.base_stats import get_total_point_pokemon
 
 
-class Home(View):
+class Home(TemplateView):
     template_name = 'battle/home.html'
 
-    def get(self, request):
-        return render(request, self.template_name)
 
+class Invite(TemplateView):
+    template_name = 'battle/invite.html'
+
+
+class BattleView(CreateView):
+    model = Battle
+    fields = ['opponent']
+
+    def form_valid(self, form):
+        form.instance.creator = self.request.user
+        form.save()
+        return HttpResponseRedirect(reverse_lazy("team_create", args=(form.instance.id,)))
+
+class TeamView(CreateView):
+    template_name = "battle/pokemon_form.html"
+    form_class = TeamForm
+    success_url = reverse_lazy("home")
+
+    def get(self, request, pk):
+        self.object = None
+        battle_object = Battle.objects.get(pk=pk)        
+        Team.objects.create(battle=battle_object, trainer=battle_object.creator)
+        Team.objects.create(battle=battle_object, trainer=battle_object.opponent)
+        return super().get(request, pk)
+
+
+    # def form_invalid(self, form):
```

Comment:
Please let's try to avoid adding new code with comments

Reply:
Sorry about it, I'm gonna delete what is not necessary